Kernel/Syscall.cpp

```diff
@@ -291,6 +291,15 @@ static u32 handle(RegisterDump& regs, u32 function, u32 arg1, u32 arg2, u32 arg3
         IO::out16(0x604, 0x2000);
         break;
     }
+    case Syscall::SC_reboot: {
```

Comment:
This looks like a reasonable way to implement reboot, but let's do it in a `Process::sys$reboot()`.
We should really only allow a superuser process with `uid=0` to reboot.
The same should be true for `halt`, I now realize. But that's a separate problem :)

Reply:
I considered permissions when adding it, I don't know why I didn't restrict it, but on second thoughts I agree that was a mistake.

Reply:
I know that the unix way is to only allow `uid=0` to reboot and halt the system? But it that really necessary here? I've always found it really strange that only a superuser has access to something so fundamental to the system (even though I _can_ kind of see the side effects of allowing a regular user to do it :P )

Reply:
It's definitely necessary. It's a Unix-like operating system, after all :)
The user can trivially acquire superuser privileges if he actually needs them or something, so preventing unprivileged processes from messing with the system is really quite cheap in terms of friction.

Reply:
The problem isn't so much the _user_ rebooting/halting things. After all, you're sitting in front of the computer, right?

The problem is that _any process you run_ can also do it. That could be pretty annoying in practice if someone gives you an auto-executing binary that just reboots you in an endless loop.

I guess another solution would be sandboxing of a sort so that certain operations like that required explicit permissions granted to them to be able to do those operations. That would be better than a user restriction, but it's also a lot more nebulous, and harder to implement right now.

Reply:
Yeah I never thought about that. Especially if there's a rogue process running that works out how to fire the syscall. Also having a multiseat system like they would at Pixar etc, where everyone's connected to the main server. I completely neglected to think that anyone can just run `reboot` and then everyone loses what they were doing....

>I guess another solution would be sandboxing of a sort so that certain operations like that required explicit permissions granted to them to be able to do those operations. That would be better than a user restriction, but it's also a lot more nebulous, and harder to implement right now.

Yeah I take it that would require a big rewrite of a fair bit of code yeah??